compiler/src/main/java/org/qbicc/graph/Deref.java

```diff
@@ -1,23 +1,40 @@
 package org.qbicc.graph;
 
+import org.qbicc.type.CompoundType;
+import org.qbicc.type.PointerType;
 import org.qbicc.type.ValueType;
 import org.qbicc.type.definition.element.ExecutableElement;
 
 /**
- * Wraps a CompoundType value which cannot be directly dereferenced. If this node is lowered an error will be thrown.
+ * Wraps a dereferenced pointer. If this node is not removed by block in the program it will be replaced by a load.
  */
 public final class Deref extends AbstractValue implements UnaryValue {
     private final Value value;
+    private final ValueHandle valueHandle;
+    private final ValueType toType;
+    private final Load load;
 
-    Deref(Node callSite, ExecutableElement element, int line, int bci, Value value) {
+    Deref(Node callSite, ExecutableElement element, int line, int bci, Node dependency, Value value) {
         super(callSite, element, line, bci);
         this.value = value;
+
+        PointerType pt = (PointerType)value.getType();
+        this.toType = pt.getPointeeType();
+
+        this.valueHandle = new PointerHandle(callSite, element, line, bci, value);
+        // compound types cannot be loaded atomically
+        this.load = new Load(callSite, element, line, bci, dependency, valueHandle,
```

Comment:
We do not want nodes to "breed" other nodes, especially `Load` given that `Deref` is not itself an `OrderedNode`.  If the `Deref` only points to a delegate `Value` then the rest of the logic should still work without the pointer handle or the load.